demosplan/DemosPlanCoreBundle/Entity/Permission/AccessControlPermission.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of the package demosplan.
+ *
+ * (c) 2010-present DEMOS plan GmbH, for more information see the license file.
+ *
+ * All rights reserved
+ */
+
+namespace demosplan\DemosPlanCoreBundle\Entity\Permission;
+
+use DateTime;
+use DemosEurope\DemosplanAddon\Contracts\Entities\AccessControlPermissionInterface;
+use DemosEurope\DemosplanAddon\Contracts\Entities\CustomerInterface;
+use DemosEurope\DemosplanAddon\Contracts\Entities\OrgaInterface;
+use DemosEurope\DemosplanAddon\Contracts\Entities\RoleInterface;
+use DemosEurope\DemosplanAddon\Contracts\Entities\UuidEntityInterface;
+use demosplan\DemosPlanCoreBundle\Entity\CoreEntity;
+use Doctrine\ORM\Mapping as ORM;
+use Gedmo\Mapping\Annotation as Gedmo;
+
+/**
+ * Stores the information about the current phase of a procedure.
+ * Currently there a two phases related to a procedure, therefore this Entity is related to the procedure twice.
+ *
+ * @ORM\Table(name="access_control_permissions")
+ *
+ * @ORM\Entity(repositoryClass="demosplan\DemosPlanCoreBundle\Repository\AccessControlPermissionRepository")
+ */
+class AccessControlPermission extends CoreEntity implements UuidEntityInterface, AccessControlPermissionInterface
+{
+    /**
+     * @ORM\Column(type="string", length=36, options={"fixed":true})
+     *
+     * @ORM\Id
+     *
+     * @ORM\GeneratedValue(strategy="CUSTOM")
+     *
+     * @ORM\CustomIdGenerator(class="\demosplan\DemosPlanCoreBundle\Doctrine\Generator\UuidV4Generator")
+     */
+    protected string $id;
+
+    /**
+     * @ORM\Column(name="permission", type="string", nullable=false)
+     */
+    protected string $permission = '';
+
+    /**
+     * @Gedmo\Timestampable(on="create")
+     *
+     * @ORM\Column(type="datetime", nullable=false)
+     */
+    private DateTime $creationDate;
+
+    /**
+     * @var DateTime
+     *
+     * @Gedmo\Timestampable(on="update")
+     *
+     * @ORM\Column(type="datetime", nullable=false)
+     */
+    private $modificationDate;
+
+    /**
+     * @var DateTime
+     *
+     * @Gedmo\Timestampable(on="update")
+     *
+     * @ORM\Column(type="datetime", nullable=false)
+     */
+    private $deletionDate;
+
+    /**
+     * @var OrgaInterface|null
+     *
+     * @ORM\ManyToOne(targetEntity="demosplan\DemosPlanCoreBundle\Entity\User\Orga")
+     *
+     * @ORM\JoinColumn(name="orga_id", referencedColumnName="_o_id", nullable=true, onDelete="SET NULL")
```

Comment:
This could be type strict 'protected ?OrgaInterface $organisation;

Reply:
This would mean there is a special (instransparent) logic: in case of null here, it means all.
Do wo rather create an entry for each orga in this case?
In case of a new orga in the DB, this one would automatically have the related permission. Is this intended?

Reply:
I would favor against an entry for any organization. This does not scale and you would need to keep it in sync with any changes of the organizations and new dynamic permissions. I would think that `null` of any of these fields mean "irrelevant for permission calculation". By default the dynamic permission is deactivated and only if a row that matches the current context is found it is enabled. When a row exists with a permission that has only nulls in the "calculation" fields (orga, customer, role, procedure) it would always be granted.

Reply:
hey @mussbach @martendemos I feel a bit lost here, how we would realize that an orga has paid and can create procedures? how would a row in this table would look like?

Reply:
@sierrazamudiodemos I thought it would be a new row with an organisation an the new permission "allowed to create procedure bla bla". Not sure if the customer and/or the role would be filled but I guess so.
That would means, that the permission is enabled for the specific orga and imply that this permission is not for all orgas (of the customer) enabled by default.

Reply:
In my mind `null` in the orga field would mean in this context for the calculation of the permissions "any organization", whereas an id would bound the permission to this specific orga. But this is open to discussion. I am with you @martendemos that both models have pros and cons